MetaMask/metamask-extension
# Add Snaps to three dots menu and refactor

## **Description**
This PR adds Snaps to the three dots menu and remove it from the Settings.

Alongside the move of the Snaps to three dots menu, refactoring of the following is done:
* Renamed routes to better describe certain Snap page locations (`/snaps`, `/snaps/view`).
* Moved all the files and folders from Settings folder to new `snaps` directory within `pages` directory.
* Renamed all the Snaps related components in order to remove all `settings-*`-like associations.
* UI in the pages for Snaps list and Snap view is done based on the Notifications page.
* Other code refactoring (CSS, locales, UI elements).
* Other minor UI changes and extended view adaptations following the Figma specifications: https://www.figma.com/file/3Z2k9kDpHxoyjwtQo1DAam/Home?node-id=76%3A13355&mode=dev
* Deprecated design system UI components are replaced with the new.
* Page component is used to wrap up the content and create consistent UI across the Snaps pages.

## **Manual testing steps**
1. Go to Settings and make sure there is nothing related to Snaps.
2. Click on the "Three dots menu" and verify that Snaps option/link is there.
3. Click on the Snaps option in the "Three dots menu" and Snaps list page should be displayed.
4. Install arbitrary snap and click on it in the Snaps list page and verify that view is displayed properly.
5. Verify both, extended view and popup.

## **Screenshots/Recordings**
### **Before**
![Screenshot 2023-10-17 at 12 30 17](https://github.com/MetaMask/metamask-extension/assets/13301024/a6349d1c-f8d9-48cc-b2b4-73bb3a933f68)
![Screenshot 2023-10-17 at 12 30 54](https://github.com/MetaMask/metamask-extension/assets/13301024/79750a13-8412-4f3e-954f-86634350b655)

### **After**
![Screenshot 2023-10-18 at 12 44 20](https://github.com/MetaMask/metamask-extension/assets/13301024/bad588b7-c801-4910-b490-e1b119a42b3b)
![Screenshot 2023-10-18 at 12 44 59](https://github.com/MetaMask/metamask-extension/assets/13301024/143f6fc9-4df2-4076-83e7-b1174e2350e5)
![Screenshot 2023-10-18 at 12 45 30](https://github.com/MetaMask/metamask-extension/assets/13301024/f159761e-face-4f3d-9141-bcd3f5fd6e5f)
Empty state:
![Screenshot 2023-10-18 at 12 48 16](https://github.com/MetaMask/metamask-extension/assets/13301024/e9756614-0fc3-405c-ab74-2d5da5d21f2e)
Hover state:
![Screenshot 2023-10-18 at 12 51 04](https://github.com/MetaMask/metamask-extension/assets/13301024/92c8a265-acc7-42f4-9a23-2448cb17e421)

## **Related issues**
Fixes: https://github.com/MetaMask/metamask-extension/issues/21168

## **Pre-merge author checklist**

## **Pre-merge reviewer checklist**
